deltacat/compute/stats/models/delta_stats.py

```diff
@@ -0,0 +1,156 @@
+                    Stats([A, D]),     #  Table 1
+                    Stats([G, J]),     #  Table 2
+                ]))
+            DeltaColumnStats("bar",
+                ManifestEntryStats([
+                    Stats([B, E]),     #  Table 1
+                    Stats([H, K]),     #  Table 2
+                ]))
+            DeltaColumnStats("baz",
+                ManifestEntryStats([
+                    Stats([C, F]),     #  Table 1
+                    Stats([I, L]),     #  Table 2
+                ]))
+        ], Stats(AllDeltaColumnStats))
+    """
+
+    @staticmethod
+    def of(columns: List[DeltaColumnStats]):
+        ds = DeltaStats()
+        ds["columns"] = columns
+        ds["stats"] = DeltaStats.generate_stats_from_columns(columns)
+        return ds
+
+    @property
+    def columns(self) -> List[DeltaColumnStats]:
+        return self["columns"]
+
+    @property
+    def stats(self) -> Optional[StatsResult]:
+        val: Dict[str, Any] = self.get("stats")
+        if val is not None and not isinstance(val, StatsResult):
+            self["stats"] = val = StatsResult(val)
+        elif val is None and self.columns:
+            self["stats"] = val = DeltaStats.generate_stats_from_columns(self.columns)
+
+        return val
+
+    @property
+    def column_names(self) -> List[str]:
+        return DeltaStats.get_column_names(self.columns)
+
+    def table_stats(self, table_idx: int) -> StatsResult:
+        return StatsResult.merge(DeltaStats.get_table_stats_list(self.columns, table_idx),
+                                 record_row_count_once=True)
+
+    def table_stats_list(self, table_idx: int) -> List[StatsResult]:
+        return DeltaStats.get_table_stats_list(self.columns, table_idx)
+
+    @staticmethod
+    def get_table_stats_list(columns: List[DeltaColumnStats], table_idx: int) -> List[StatsResult]:
+        """
+        Helper method to provide a list of columnar stats for a specific table/manifest entry
+
+        TODO (ricmiyam): A table specific stats container would be nice to have (i.e. TableStats, TableColumnStats)
+        """
+        dataset_columnar_stats_list: List[ManifestEntryStats] = [column.manifest_stats for column in columns
+                                                                 if column.manifest_stats is not None]
+        try:
+            return [stats.stats[table_idx] for stats in dataset_columnar_stats_list]
+        except IndexError:
+            sci: ManifestEntryStats = dataset_columnar_stats_list[0]
+            raise ValueError(f"Table index {table_idx} is not present in this dataset of {sci.delta_locator} "
+                             f"with manifest table count of {len(sci.stats)}")
+
+    @staticmethod
+    def get_column_names(columns: List[DeltaColumnStats]):
+        return [column_stats.column for column_stats in columns] if columns else []
+
+    @staticmethod
+    def generate_stats_from_columns(columns: List[DeltaColumnStats],
```

Comment:
For consistency w/ other static method naming conventions and since it seems like you're fundamentally using this to get full delta-level stats:
```suggestion
    def get_delta_stats(columns: List[DeltaColumnStats],
```